src/ch.usi.inf.nodeprof/src/ch/usi/inf/nodeprof/analysis/ProfilerExecutionEventNode.java

```diff
@@ -135,10 +136,16 @@ protected void onReturnExceptional(VirtualFrame frame, Throwable exception) {
         if (!profilerEnabled) {
             return;
         }
+
         if (hasOnEnter > 0) {
             hasOnEnter--;
             this.cb.exceptionHitCount++;
-            this.child.executeExceptional(frame, exception);
+            if (exception instanceof ControlFlowException) {
+                Object[] inputs = child.expectedNumInputs() != 0 ? getSavedInputValues(frame) : null;
+                this.child.executeExceptionalCtrlFlow(frame, exception, inputs);
```

Comment:
Do you think it would be necessary to always provide inputs for ```executeExceptional```. Instead of having a check of exception type here, each event node simply deals with the possible exception it could face with the input values.

Reply:
Consistently passing the inputs might make sense. 

But the main reason for the change was that `ControlFlowException`s are kind of a special (Truffle-specific) kind of exception that should probably be treated differently from a program exception in the NodeProf handlers. Right now, some (non-control-flow) handlers don't override the exception case, but if the would, they would always(?) need to handle `ControlFlowException` as not-a-real-exception.